### PR TITLE
Add loading state and error message to Deployment map

### LINKED ISSUE #703: Add loading state and error message to Deployment map
> I was thinking about adding a load state to the Deployment map. I've noticed that when we access a URL directly to that section, like http://admg-inventory-staging.s3-website-us-east-1.amazonaws.com/casei/campaign/ARCTAS/#deployment, there is no indication that the geojson data is loading, and it can make some users abandon the page.

> What do you think? Maybe we would also need a message to display when the map data is not available?

_Originally posted by @willemarcel in https://github.com/NASA-IMPACT/admg-casei/issues/701#issuecomment-2443318407_
            
- [ ] add a load state to the Deployment map
- [ ] display message when the map data is not available

### PR DESCRIPTION
**Pull request by willemarcel**
Resolves #703

@LanesGood feel free to modify the style or text.

### CODE REVIEW

File: src/components/timeline/timeline-chart.js [hunk 1]
```diff
@@ -145,7 +130,18 @@ export const TimelineChart = ({ deployments, bounds, campaignName }) => {
 
   return (
     <Disclosure open={!!selectedDeployment}>
-      {geojson?.features?.length && (
+      {geojsonError && (
+        <MapErrorMsg>
+          <ExclamationIcon color={colors.darkTheme.highlight} />
+          <h4>It was not possible to load the map data for this campaign.</h4>
```
Comment (LanesGood) [inline src/components/timeline/timeline-chart.js]:
```suggestion
          <h4>There was an issue loading the map data for this campaign</h4>
```

**Review by LanesGood — CHANGES_REQUESTED** [review]
@willemarcel I think this is looking good, but I'm seeing the error message on every campaign that doesn't have flight data. If I understand the code correctly, this should only appear if there is flight data, but it errors - is that correct? Should there be a `geojson?.features?.length` check combined with the `geojsonError` check?


**EDIT**
I attempted this, and then intentionally malformed the geojson for a campaign, but now I simply don't see the deployment map and there's no error message:
image

I also thought the loader should come first.

**Comment by willemarcel** [comment]
Actually, the only error that can happen is that we don't have a geojson for the campaign. Should we modify the text to state more clearly that it wasn't an error, but the map data for that campaign is not available?

When running locally, you need to enable throttling in your network console to see the loading map component.

File: src/components/timeline/timeline-chart.js (hunk 1 above)
Comment (LanesGood) [reply src/components/timeline/timeline-chart.js]: @willemarcel in the case you've suggested that no errors beyond missing data are present, I would suggest we don't call this an error and use a different icon. 

Lets state "Flight path data is not yet available for this campaign" if that is true.

**Comment by LanesGood** [comment]
@willemarcel looking again at this, I understand my confusion and think there needs to be some slight refactoring. The `DeploymentMap` is currently a child of the `TimelineChart` component.  In `TimelineChart` the whole component is wrapped in a ReachUI `Disclosure` component to hide or show the timeline. This is behavior is unnecessary to wrap around the Deployment Map.

Instead, I would suggest the following:

- Rename `TimelineSection` to `DeploymentEventsSection`
- Rename `Timeline` directory within "Components" to `DeploymentsEvents`
- Within this new `DeploymentEventsSection`, nest the existing `TimelineChart` and add a new child component, `DeploymentMap`
- Move the existing `MapErrorMsg` and `MapLoading` components to the existing `DeploymentMap` to contain those concerns, and move the geojson fetching/validation logic to that as well

**Comment by LanesGood** [comment]
New message and icon for empty geojson:
image

Should this be called `MapErrorMsg` or `MapEmptyStateMsg`? Bigger picture, is there a need to call the `error` type of the useFetch hook just for empty data, or could we simply set this message if the `geoJson` response returns `null`, and use a different/more accurate message if there is a true error?

**Comment by willemarcel** [comment]
> Should this be called `MapErrorMsg` or `MapEmptyStateMsg`?

I like `MapEmptyStateMsg`

> Bigger picture, is there a need to call the `error` type of the useFetch hook just for empty data, or could we simply set this message if the `geoJson` response returns `null`, and use a different/more accurate message if there is a true error?

What we can do instead is check the status code of the request. If it's 404, the geojson file of that campaign was not found.

**Review by LanesGood — APPROVED** [review]

**Pull request merged**